source/s3_client.c

```diff
@@ -80,6 +81,12 @@ static size_t s_dns_host_address_ttl_seconds = 5 * 60;
  * 30 seconds mirrors the value currently used by the Java SDK. */
 static const uint32_t s_default_throughput_failure_interval_seconds = 30;
 
+/* Default multiplier between max part size and memory limit */
+static const size_t s_default_max_part_size_to_mem_lim_multiplier = 4;
```

Comment:
style: there's a relationship between this magic number, and the `buffer_pool->primary_size_cutoff = chunk_size * 4;` magic number from s3_buffer_pool.c

Add comment about this relationship

Or is it worth having 1 magic number in s3_buffer_pool.c, and common function both can call, like `aws_s3_buffer_pool_get_primary_size_cutoff(size_t chunk_size)`? Or first we create the buffer_pool, then s3_client.c calls a getter to clamp the client->max_part_size if necessary, like:
```
client->max_part_size = aws_size_min(client->max_part_size, aws_s3_buffer_pool_get_primary_size_cutoff(client->buffer_pool));
```